src/controllers/csi/gc/controller.go

```diff
@@ -132,17 +142,31 @@ func collectGCInfo(ctx context.Context, apiReader client.Reader, dynakube dynatr
 	}, nil
 }
 
+func isSafeToGC(access metadata.Access, dynakubeList *dynatracev1beta1.DynaKubeList) bool {
+	dkMetadataList, err := access.GetAllDynakubes()
+	if err != nil {
+		log.Info("failed to get dynakube metadata from database, err: %s")
+		return false
+	}
+	filteredDynakubes := filterCodeModulesImageDynakubes(dynakubeList)
+	for _, dkMetadata := range dkMetadataList {
+		if dkMetadata.LatestVersion == "" {
```

Comment:
```suggestion
		if isDownloading(dkMetadata.LatestVersion) {
		
func isDownloading(latestVersion string) bool {
  return  latestVersion == ""
}
```

would make it clear why it is not safe to gc in that scenario